runtime/src/bank.rs

```diff
@@ -2794,7 +2794,27 @@ impl Bank {
                         }
                     };
                     if cached_stake_account != &stake_account {
-                        invalid_cached_stake_accounts.fetch_add(1, Relaxed);
+                        let mut cached_account = cached_stake_account.account().clone();
+                        // We could have collected rent on the loaded account already in this new epoch (we could be at partition_index 12, for example).
+                        // So, we may need to adjust the rent_epoch of the cached account. So, update rent_epoch and compare just the accounts.
```

Comment:
Can you please point me to where in the code this rent collections is done and accounts are updated in accounts-db?
Shouldn't we there update the sakes-cache as well?